pkg/manager/impl/execution_manager.go

```diff
@@ -292,17 +336,34 @@ func setCompiledTaskDefaults(ctx context.Context, config runtimeInterfaces.Confi
 	if resource != nil && resource.Attributes != nil && resource.Attributes.GetTaskResourceAttributes() != nil {
 		taskResourceSpec = resource.Attributes.GetTaskResourceAttributes().Defaults
 	}
-	task.Template.GetContainer().Resources.Requests = assignResourcesIfUnset(
+	var resourceRequestIndices = newTaskResourceIndices()
+	task.Template.GetContainer().Resources.Requests, resourceRequestIndices = assignResourcesIfUnset(
 		ctx, task.Template.Id, config.TaskResourceConfiguration().GetDefaults(), task.Template.GetContainer().Resources.Requests,
 		taskResourceSpec)
 
 	logger.Debugf(ctx, "Assigning task resource limits for [%+v]", task.Template.Id)
 	if resource != nil && resource.Attributes != nil && resource.Attributes.GetTaskResourceAttributes() != nil {
 		taskResourceSpec = resource.Attributes.GetTaskResourceAttributes().Limits
 	}
-	task.Template.GetContainer().Resources.Limits = assignResourcesIfUnset(
+
+	var resourceLimitIndices = newTaskResourceIndices()
+	task.Template.GetContainer().Resources.Limits, resourceLimitIndices = assignResourcesIfUnset(
 		ctx, task.Template.Id, createTaskDefaultLimits(ctx, task), task.Template.GetContainer().Resources.Limits,
 		taskResourceSpec)
+
+	if !(resourceRequestIndices.isValid() && resourceLimitIndices.isValid()) {
+		logger.Warningf(ctx, "Validating task resource requests & limits that are incomplete [%+v]", task.Template.Id)
```

Comment:
This warning message is a present progressive so the meaning is a bit unclear to me. Is it reporting what it is trying to do? Shouldn't it report that those indices are invalid?